src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1415,15 +1415,83 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
 
         case NI_Vector128_ConvertToDouble:
         case NI_Vector256_ConvertToDouble:
+        case NI_Vector512_ConvertToDouble:
+        {
+            assert(sig->numArgs == 1);
+            assert(varTypeIsLong(simdBaseType) || simdBaseType == TYP_FLOAT);
+            if (IsBaselineVector512IsaSupportedOpportunistically())
+            {
+                if (varTypeIsLong(simdBaseType))
+                {
+                    intrinsic = (simdSize == 16) ? NI_AVX512DQ_VL_ConvertToVector128Double
+                                                 : (simdSize == 32) ? NI_AVX512DQ_VL_ConvertToVector256Double
+                                                                    : NI_AVX512DQ_ConvertToVector512Double;
+                }
+                else
+                {
+                    intrinsic = (simdSize == 16) ? NI_SSE2_ConvertToVector128Double
+                                                 : (simdSize == 32) ? NI_AVX_ConvertToVector256Double
+                                                                    : NI_AVX512F_ConvertToVector512Double;
+                }
```

Comment:
I don't think this path can be hit. We only have managed APIs for `long` and `ulong` to `double` exposed

Reply:
Change made.